File: contest_template/cpp/tester.cpp
#include <bits/stdc++.h>
using namespace std;

class JudgeInterface {
private:
    int query_count;
    int max_queries;

public:
    JudgeInterface() : query_count(0), max_queries(100) {} // デフォルトのクエリ制限

    void set_max_queries(int max) {
        max_queries = max;
    }

    void output(const string& message) {
        cout << message << endl;
        cout.flush();
    }

    string input() {
        if (query_count >= max_queries) {
            wrong_answer("クエリ制限超過");
        }
        query_count++;
        
        string input;
        getline(cin, input);
        return input;
    }

    void correct_answer() {
        exit(0);
    }

    void wrong_answer(const string& message) {
        cerr << "Wrong Answer: " << message << endl;
        exit(1);
    }
};

int main() {
    JudgeInterface judge;
    // TODO: インタラクティブジャッジの実装
    // 例: 数当てゲーム
    int n = 42;
    judge.output("数当てゲームを開始します。1から100の数字を当ててください。");
    
    while (true) {
        string input = judge.input();
        try {
            int guess = stoi(input);
            if (guess == n) {
                judge.output("正解です！");
                judge.correct_answer();
            } else if (guess < n) {
                judge.output("もっと大きいです");
            } else {
                judge.output("もっと小さいです");
            }
        } catch (...) {
            judge.wrong_answer("不正な入力です");
        }
    }
    return 0;
} 

File: contest_template/rust/src/tester.rs
use std::io::{self, Write};

struct JudgeInterface {
    query_count: u32,
    max_queries: u32,
}

impl JudgeInterface {
    fn new() -> Self {
        Self {
            query_count: 0,
            max_queries: 100, // デフォルトのクエリ制限
        }
    }

    fn set_max_queries(&mut self, max_queries: u32) {
        self.max_queries = max_queries;
    }

    fn output(&self, message: &str) {
        println!("{}", message);
        io::stdout().flush().unwrap();
    }

    fn input(&mut self) -> String {
        if self.query_count >= self.max_queries {
            self.wrong_answer("クエリ制限超過");
        }
        self.query_count += 1;
        
        let mut input = String::new();
        io::stdin().read_line(&mut input).unwrap();
        input.trim().to_string()
    }

    fn correct_answer(&self) -> ! {
        std::process::exit(0);
    }

    fn wrong_answer(&self, message: &str) -> ! {
        eprintln!("Wrong Answer: {}", message);
        std::process::exit(1);
    }
}

fn main() {
    let mut judge = JudgeInterface::new();
    // TODO: インタラクティブジャッジの実装
    // 例: 数当てゲーム
    let n = 42;
    judge.output("数当てゲームを開始します。1から100の数字を当ててください。");
    
    loop {
        let input = judge.input();
        match input.parse::<i32>() {
            Ok(guess) => {
                if guess == n {
                    judge.output("正解です！");
                    judge.correct_answer();
                } else if guess < n {
                    judge.output("もっと大きいです");
                } else {
                    judge.output("もっと小さいです");
                }
            }
            Err(_) => judge.wrong_answer("不正な入力です"),
        }
    }
} 